src/main/java/algorithms/leetcode/medium/ReverseLinkedList2.java

```diff
@@ -0,0 +1,71 @@
+package algorithms.leetcode.medium;
+/*
+92. Reverse Linked List II
+MEDIUM
+
+Given the head of a singly linked list and two integers left and right where left <= right, reverse the nodes of the list
+from position left to position right, and return the reversed list.
+
+Example 1:
+Input: head = [1,2,3,4,5], left = 2, right = 4
+Output: [1,4,3,2,5]
+
+Example 1:
+Input: head = [5], left = 1, right = 1
+Output: [5]
+
+Constraints:
+The number of nodes in the list is n.
+1 <= n <= 500
+-500 <= Node.val <= 500
+1 <= left <= right <= n
+
+ */
+
+class ListNode {
+    int val;
+    ListNode next;
+
+    ListNode() {
+    }
+
+    ListNode(int val) {
+        this.val = val;
+    }
+
+    ListNode(int val, ListNode next) {
+        this.val = val;
+        this.next = next;
+    }
+}
+
+public class ReverseLinkedList2 {
```

Comment:
Not able to understand this logic. Why can't we do similar to the other reverse linked list problem? We have start and end nodes.

Reply:
Reversing logic is same but here we are finding the starting first (using 1st while) and then reversing it (in 2nd while loop). 
This particular sum is in the next [PR](https://github.com/Sreedevi-27/Problem-Solving/pull/13/commits/3ef8710a2811c46bc66875215bbee73f9e1561b6#diff-b1aa6470b7deb654c328d280c95d0c1fd39cc500d3626088af52387661d92416). Added this in this PR by mistake.